packages/composite-checkout/src/components/checkout-steps.tsx

```diff
@@ -31,16 +33,45 @@ const debug = debugFactory( 'composite-checkout:checkout-steps' );
 
 const customPropertyForSubmitButtonHeight = '--submit-button-height';
 
+// See https://github.com/reactjs/rfcs/blob/useevent/text/0000-useevent.md
+function useEvent< A, R >( handler: ( ...args: A[] ) => R ): ( ...args: A[] ) => R {
```

Comment:
We talked about this on a call, but for posterity I'll also note it here: we might consider using a different name for our local version of this hook. If/when the react team add their proposed `useEvent`, there may be subtle reasons why it can't be a drop in replacement and the "real" `useEvent` has to coexist with this one in the code while we fix that. In that case there's a risk of someone later reading this name and thinking it's react's `useEvent`. Not a huge risk, but one that's easily avoided with a different name.

We talked about a couple of alternatives like `useCallbackWithEffects` and `useJITCallback`; I don't have strong feelings about which to use.

Reply:
Ended up going with useJITCallback with a bigger docblock to explain its purpose.